Brass knuckles: finally working edition (now with rebalances and more fun)

c420-o opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
I made this like three times but the other ones were on outdated and fucked up repos. this one works like it's supposed to, it's tested, rebalanced, sprited, cleaned up the code, et cetera. 
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
there is no longer a continuity error with the "punch shit" poster. now brass knuckles are canon, and they're balanced so you can fuck people up without being too overpowered, if that makes sense. merge if you wanna be a badass and punch shit. 
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Adds the brass knuckles
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

c420-o committed:
I just made a new repo and added all the other shit and it works now

I don't know why this was a struggle for me

c420-o commented:
I swear this is the last time I'll make a seperate repo for a fucked up master

interestingusernam3 commented on code/datums/martial/brassknuckles.dm:
```diff
@@ -0,0 +1,35 @@
+					"<span class='userdanger'>You're [picked_hit_type]ed by [A]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", COMBAT_MESSAGE_RANGE, A)
+	to_chat(A, "<span class='danger'>You [picked_hit_type] [D]!</span>")
+	log_combat(A, D, "punched")
+	D.apply_damage(rand(5,10), BRUTE, affecting, armor_block, wound_bonus = 15)
```
This on average deals less damage than a scalpel while having the same wound_bonus, FYI.

c420-o replied:
Yes but the difference is it's blunt wounds so all it does is break bones

ATH1909 commented on code/datums/martial/brassknuckles.dm:
```diff
@@ -0,0 +1,35 @@
+
+/obj/item/clothing/gloves/brassknuckles
+	name = "Brass Knuckles"
+	desc = "a contraband item meant to prove weapons are for pussies. Perfect for any bloodthirsty soldier or lunatics wearing tiger masks."
```
```suggestion
	desc = "A contraband item meant to prove that weapons are for pussies. Perfect for any bloodthirsty soldier or lunatics wearing tiger masks."
```

c420-o replied:
Sure why not

c420-o committed:
Update code/datums/martial/brassknuckles.dm

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

c420-o commented:
Wait I fucked up

necromanceranne commented:
PEople webedit all the time. It's when you webedit EVERY PR.

Twaticus reviewed (changes requested):
![chrome_7vMzLeoVw0](https://user-images.githubusercontent.com/46540570/107720687-d5ab2200-6ca8-11eb-9980-07c2d1b31c16.png)
im sorry but nothing about this says brass knuckles

Rohesie commented:
Very much not a fan of this martial art.

Also, won't the user retain it by putting it on the gloves slot and then moving it to the backpack without dropping it?

c420-o commented:
> ![chrome_7vMzLeoVw0](https://user-images.githubusercontent.com/46540570/107720687-d5ab2200-6ca8-11eb-9980-07c2d1b31c16.png)
> im sorry but nothing about this says brass knuckles

yeah I know. I'll def change the icons



> Very much not a fan of this martial art.
> 
> Also, won't the user retain it by putting it on the gloves slot and then moving it to the backpack without dropping it?

no?

`/obj/item/clothing/gloves/brassknuckles/dropped(mob/user)
	. = ..()
	if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src)
		style.remove(user)`

Rohesie commented:
> no?
> 
> `/obj/item/clothing/gloves/brassknuckles/dropped(mob/user) . = ..() if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src) style.remove(user)`

Is this called when the item goes back from the gloves slot into the hands slot, and then into the backpack?

c420-o commented:
> > no?
> > `/obj/item/clothing/gloves/brassknuckles/dropped(mob/user) . = ..() if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src) style.remove(user)`
> 
> Is this called when the item goes back from the gloves slot into the hands slot, and then into the backpack?

it...gets called when the gloves aren't in the slot...

c420-o committed:
updated icon so it looks acceptable

better than the last one

c420-o commented:
![image](https://user-images.githubusercontent.com/76887567/107727925-8c63ce00-6cba-11eb-966d-a78bdc3426c6.png)
new half-assed inhand icon

Rohesie commented:
> 
> 
> > > no?
> > > `/obj/item/clothing/gloves/brassknuckles/dropped(mob/user) . = ..() if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src) style.remove(user)`
> > 
> > 
> > Is this called when the item goes back from the gloves slot into the hands slot, and then into the backpack?
> 
> you can't move gloves from the glove slot directly into the backpack

Do please re-read what I wrote.

c420-o commented:
> > 
> > 
> > > > no?
> > > > `/obj/item/clothing/gloves/brassknuckles/dropped(mob/user) . = ..() if(user.get_item_by_slot(ITEM_SLOT_GLOVES) == src) style.remove(user)`
> > > 
> > > 
> > > Is this called when the item goes back from the gloves slot into the hands slot, and then into the backpack?
> > 
> > you can't move gloves from the glove slot directly into the backpack
> 
> Do please re-read what I wrote.

Alright I see what you're saying. It's copy pasted krav maga code, so it's unlikely but I'll still test it

c420-o commented:
So after doing some testing I realized that removing the gloves actually don't remove harm_act. this isn't an issue with krav maga gloves for some reason, So I made it that harm_act is removed as well

c420-o committed:
improves code

adds check streak so A and D are defined, and style variable is added, so now it the style can be forgotten.

c420-o reviewed (commented):
it's all good to go!

Rohesie commented:
It's not a good solution to check every time you harm, instead we need to guarantee it gets properly added and removed when the conditions are met.

However, I'll be very honest with you. I dislike the idea of expanding martial arts. They're not in a good shape code-wise right now. So unless you intend to do an out-of-scope big refactor for them, I don't think this PR is going far.

c420-o commented:
> instead we need to guarantee it gets properly added and removed when the conditions are met.

Well style does that, the harm is just a double check thing. Better to be safe than sorry

>I don't think this PR is going far.

:(((

c420-o committed:
removes harm_act check and checkstreak

Rohesie commented:
I'm very sorry. I've been thinking this over and I'd really rather `/datum/martial_art` wasn't expanded.
Ideally we refactor this before.

Do you want to rework your PR to use some other system, or simply close it?
I leave it up to you, I have no strong opinions on the feature itself, the addition of brass knuckles, though I'd prefer new abilities or weapons were more visible whenever not intended to be used stealthily.

c420-o commented:
> I'm very sorry. I've been thinking this over and I'd really rather `/datum/martial_art` wasn't expanded.
> Ideally we refactor this before.
> 
> Do you want to rework your PR to use some other system, or simply close it?
> I leave it up to you, I have no strong opinions on the feature itself, the addition of brass knuckles, though I'd prefer new abilities or weapons were more visible whenever not intended to be used stealthily.

I'm gonna try to rework it

TheBonded commented:
damn
I really wanted this to be added
